inductiva/projects/project.py

```diff
@@ -0,0 +1,102 @@
+"""Project class"""
+import contextvars
+
+import inductiva
+from inductiva.client import ApiException
+from inductiva.client.apis.tags import projects_api
+
+CURRENT_PROJECT = contextvars.ContextVar("current_project", default=None)
+
+
+def get_current_project():
+    """Gets the current project"""
+    return CURRENT_PROJECT.get()
+
+
+class Project:
+    """Projects class.
+
+    This class manages the cuurent project being used. Which has
+    implications to where tasks are submitted.
+
+    Example usage:
+
+    >>> project = inductiva.projects.Project("test_project")
+    >>> project.start()
+    >>> simulator.run(...)  # Submitted to  `'test_project'`
+    >>> project.stop()
+
+    This is equivalent to:
+
+    >>> with inductiva.projects.Project("test_project"):
+    >>>    simulator.run(...)
+
+    """
+
+    AVAILABLE_MODES = ["r", "w"]
+
+    def __init__(self, name: str, exists_ok: bool = True, mode: str = "w"):
```

Comment:
`error_on_exists` might be a better choice than `exists_ok`.

Reply:
`exists_ok` is similar to `os.makedirs(exists_ok)`. That is why I chose the variable name. Keeps things intuitive with the `os` library. But now that I am looking at it I think that we should actually make it default to false.